### PR TITLE
Identity abstraction and PrivateKey

### LINKED ISSUE #26: [SDK][Trade] Identity
In order to support many different way of signing (single private key, mnemonic, injected browser extensions, hardware wallets) we need to introduce an abstraction for the `Wallet` that can use to sign a given swap PSET.

I propose to pass the following object to the SDK.Trade constructor (could be an interface)

```js
identity : {
   // `key`,`mnemonic`,`inject`, `ledger`, `trezor`
  type: `string`,
  value: `string|Function|InjectIdentity`
  network: `liquid.Network`
}
```

### PR DESCRIPTION
**Pull request by tiero**
This commit add an abstraction to handle many different ways of signing. Is not complete, but is a basic Identity interface tha defines the method needed such as signPset and blindPset. Note that inside Trade instance, the walle﻿t will use the Identity class to sign and blind.

Please @louisinger keep an eye on it

It closes #26 
It closes https://github.com/TDex-network/tdex-sdk/issues/32
It closes https://github.com/TDex-network/tdex-sdk/issues/33

TODOS

- [x] Base identity class and types
- [x] PrivateKey implementation

### CODE REVIEW

File: src/identities/privatekey.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    // sign all the inputs asynchronously
+    await Promise.all(
+      pset.data.inputs.map((_, index: number) =>
+        pset.signInputAsync(index, this.signingKeyPair)
+      )
+    );
+
+    // validate all the signature
```
Comment (louisinger) [inline src/identities/privatekey.ts:94]: I've commented on this part, don't know if it's necessary. @tiero

File: src/identities/privatekey.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    // create payment
+    const p2wpkh = payments.p2wpkh({
+      pubkey: this.signingKeyPair.publicKey,
+      blindkey: this.blindingKeyPair.publicKey,
+      network: this.network,
+    });
+
+    // store data inside private fields.
+    this.address = p2wpkh.confidentialAddress!;
+    this.blindPrivKey = this.blindingKeyPair.privateKey!.toString('hex');
```
Comment (tiero) [inline src/identities/privatekey.ts:64]: There is no need to duplicate here. We could always encode to hex inline when needed from this.blindingKeyPair

File: src/identities/privatekey.ts (hunk 1 above)
Comment (tiero) [reply src/identities/privatekey.ts:94]: Nope, we leave validation out of this scope. Please remove it.

File: src/identities/privatekey.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    // }
+    // return the base64 encoded pset.
+    return pset.toBase64();
+  }
+
+  /**
+   * for private key: only returns one confidential address & the associated blindingPrivKey.
+   */
+  getAddresses(): AddressInterface[] {
+    return [{ address: this.address, blindPrivKey: this.blindPrivKey }];
```
Comment (tiero) [inline src/identities/privatekey.ts:120]:
```suggestion
    return [{ 
      confidentialAddress: this.address, 
      blindingPrivateKey: this.blindingKeyPair.privateKey!.toString('hex') 
    }];
```

Comment (tiero) [reply src/identities/privatekey.ts:120]: Better make the name of returned keys more explicit, even if the name are longer.

**Comment by louisinger** [comment]
For me, the merge commit breaks two `swap.test.ts`, that's strange because the test passes on Github.

File: tsconfig.json [hunk 4]
```diff
@@ -1,6 +1,7 @@
 {
   "include": [
-    "src",
+    "src", 
+    "src/identities/*.ts",
```
Comment (tiero) [inline tsconfig.json:4]: why this necessary? Would be better `src/*`

File: tsconfig.json [hunk 5]
```diff
@@ -15,7 +16,6 @@
     "importHelpers": true,
     "declaration": true,
     "sourceMap": true,
-    "rootDir": "./src",
```
Comment (tiero) [inline tsconfig.json:18]: maybe `./src/**/*`

File: tsconfig.json (hunk 4 above)
Comment (louisinger) [reply tsconfig.json]: Indeed it's not necessary: `src` seems to work

File: tsconfig.json (hunk 5 above)
Comment (louisinger) [reply tsconfig.json:18]: I want to include the test folder too

**Pull request merged**